distribution/pull.go

```diff
@@ -165,11 +168,11 @@ func Pull(ref reference.Named, imagePullConfig *ImagePullConfig) error {
 // status message indicates that a newer image was downloaded. Otherwise, it
 // indicates that the image is up to date. requestedTag is the tag the message
 // will refer to.
-func writeStatus(requestedTag string, out io.Writer, sf *streamformatter.StreamFormatter, layersDownloaded bool) {
+func writeStatus(requestedTag string, out chan<- xfer.Progress, layersDownloaded bool) {
 	if layersDownloaded {
-		out.Write(sf.FormatStatus("", "Status: Downloaded newer image for %s", requestedTag))
+		out <- xfer.Progress{Message: "Status: Downloaded newer image for " + requestedTag}
 	} else {
-		out.Write(sf.FormatStatus("", "Status: Image is up to date for %s", requestedTag))
+		out <- xfer.Progress{Message: "Status: Image is up to date for " + requestedTag}
```

Comment:
From client side it would be same that I described in https://github.com/aaronlehmann/docker/pull/5#discussion_r45641723 . No preference if it is a callback or a channel, but read and write ends don't have to be same objects. And in callback there could also be a different object that doesn't require ID.